src/query-builder/case-builder.ts

```diff
@@ -0,0 +1,162 @@
+import { Expression } from '../expression/expression.js'
+import { ExpressionWrapper } from '../expression/expression-wrapper.js'
+import { freeze } from '../util/object-utils.js'
+import {
+  ExtractTypeFromReferenceExpression,
+  SimpleReferenceExpression,
+  parseReferenceExpression,
+} from '../parser/reference-parser.js'
+import { CaseNode } from '../operation-node/case-node.js'
+import { parseExpression } from '../parser/expression-parser.js'
+import { WhenNode } from '../operation-node/when-node.js'
+
+export class CaseBuilder<DB, TB extends keyof DB, W = unknown, O = never>
+  implements Whenable<DB, TB, W, O>
+{
+  readonly #props: CaseBuilderProps
+
+  constructor(props: CaseBuilderProps) {
+    this.#props = freeze(props)
+  }
+
+  when(expression: Expression<W>): CaseThenBuilder<DB, TB, W, O> {
```

Comment:
Wondering if we should also have a shortcut for `cmpr`: `when('a', 'op', 'b')`? Thoughts?

Reply:
Great idea!

I'm also adding the single value variant (case...value...when...then...end).